sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2422,7 +2422,7 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       Row(s"$expected") :: Nil)
   }
 
-  test("SPARK-15752 optimize metadata only query for datasource table") {
+  ignore("SPARK-15752 optimize metadata only query for datasource table") {
```

Comment:
This is kind of aggressive.  I can revert this one if anyone objects.

Reply:
I object. You are removing integration testing for whomever it using the feature.

Reply:
we should either fix the rule, or remove it completely if it has fundamental issues. I don't think the Hive approach is the right thing to do: we should not leave a problematic rule that users can enable.